File: android/app/src/main/java/com/drivers/test/view/screen/HomeScreen.kt
package com.drivers.test.view.screen

import android.content.Intent
import android.net.Uri
import androidx.activity.compose.BackHandler
import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.expandVertically
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.shrinkVertically
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.drivers.test.model.QuizMode
import com.drivers.test.model.StateInfo
import com.drivers.test.theme.AppTheme
import com.drivers.test.view.components.LanguageBar
import com.drivers.test.view.components.PrimaryButton
import com.drivers.test.view.components.StatItem
import com.drivers.test.viewmodel.QuizViewModel

@Composable
fun HomeScreen(vm: QuizViewModel) {
    BackHandler { vm.goStatePicker() }

    val c = AppTheme.colors
    val state = vm.currentState ?: return

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
            .padding(top = 24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        LanguageBar(
            vm.currentLang,
            langs = state.languages.ifEmpty { listOf("en", "es") },
            langLabels = vm.localizer.langLabels,
            officialTestLanguages = state.officialTestLanguages,
        ) { vm.switchLang(it) }

        state.officialTestLanguages?.let {
            Spacer(Modifier.height(4.dp))
            Text(
                vm.t("officialTestCaption", mapOf("agency" to state.agency)),
                fontSize = 11.sp,
                color = c.gray,
            )
        }

        Spacer(Modifier.height(16.dp))

        Text(
            vm.t("title", mapOf("state" to state.code.uppercase(), "state_name" to state.name)),
            fontSize = 28.sp,
            fontWeight = FontWeight.Bold,
            color = c.blue,
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(8.dp))
        Text(
            vm.t("subtitle", mapOf("state_name" to state.name, "agency" to state.agency)),
            fontSize = 15.sp,
            color = c.gray,
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(4.dp))
        Text(
            vm.t(
                "passingScore",
                mapOf(
                    "pass_pct" to "${state.passingScorePct}",
                    "pass_count" to "${state.passCount}",
                    "test_count" to "${state.testQuestionCount}",
                ),
            ),
            fontSize = 13.sp,
            color = c.gray,
        )

        Spacer(Modifier.height(16.dp))

        if (state.source != null || state.verification != null || state.categories != null) {
            AboutThisTestSection(state, vm)
            Spacer(Modifier.height(16.dp))
        }

        // Stats banner
        val history = vm.quizHistory()
        if (history.isNotEmpty()) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .background(c.card)
                    .border(1.dp, MaterialTheme.colorScheme.outline, RoundedCornerShape(12.dp))
                    .padding(14.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                    StatItem("${history.size}", vm.t("quizzes"))
                    StatItem("${vm.averageScore()}%", vm.t("avgScore"))
                    StatItem("${vm.passStreak()}", vm.t("passStreak"))
                }
                Text(
                    vm.t("viewStats"),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = c.blue,
                    modifier = Modifier.clickable { vm.showStats() },
                )
            }
            Spacer(Modifier.height(16.dp))
        }

        // Mode selector
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            ModeButton(
                title = vm.t("modeRandom"),
                desc = vm.t("modeRandomDesc"),
                icon = Icons.Filled.Refresh,
                isActive = vm.quizMode == QuizMode.RANDOM,
                modifier = Modifier.weight(1f),
            ) { vm.quizMode = QuizMode.RANDOM }

            ModeButton(
                title = vm.t("modeWeak"),
                desc = vm.t("modeWeakDesc"),
                icon = Icons.Filled.Star,
                isActive = vm.quizMode == QuizMode.WEAK,
                badgeCount = vm.weakQuestions().size,
                modifier = Modifier.weight(1f),
            ) { vm.quizMode = QuizMode.WEAK }
        }

        Spacer(Modifier.height(16.dp))

        Text(vm.t("numQuestions"), fontSize = 15.sp, fontWeight = FontWeight.SemiBold)
        Spacer(Modifier.height(8.dp))

        // Count selector
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            vm.countOptions().forEach { count ->
                val isAll = count == state.totalQuestions
                val isSelected = vm.selectedCount == count
                Text(
                    text = if (isAll) vm.t("all") else "$count",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = if (isSelected) c.onPrimary else c.blue,
                    style = TextStyle(fontFeatureSettings = "tnum"),
                    modifier = Modifier
                        .defaultMinSize(minHeight = 44.dp)
                        .clip(RoundedCornerShape(12.dp))
                        .background(if (isSelected) c.blue else c.card)
                        .border(2.dp, c.blue, RoundedCornerShape(12.dp))
                        .clickable { vm.selectedCount = count }
                        .padding(horizontal = 20.dp, vertical = 10.dp)
                        .wrapContentSize(Alignment.Center),
                )
            }
        }

        Spacer(Modifier.height(20.dp))

        val weakEmpty = vm.quizMode == QuizMode.WEAK && vm.weakQuestions().isEmpty()
        PrimaryButton(
            text = if (weakEmpty) vm.t("noWeakSpots") else vm.t("startQuiz"),
            enabled = !weakEmpty,
        ) { vm.startQuiz() }

        Spacer(Modifier.height(10.dp))

        Text(
            vm.t("changeState"),
            fontSize = 14.sp,
            color = c.gray,
            modifier = Modifier.clickable { vm.goStatePicker() }.padding(10.dp),
        )
    }
}

@Composable
private fun AboutThisTestSection(
    state: StateInfo,
    vm: QuizViewModel,
) {
    var expanded by remember { mutableStateOf(false) }
    val c = AppTheme.colors
    val chevronRotation by animateFloatAsState(if (expanded) 180f else 0f, label = "chevron")
    val context = LocalContext.current

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(c.card)
            .border(1.dp, MaterialTheme.colorScheme.outline, RoundedCornerShape(12.dp)),
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable { expanded = !expanded }
                .padding(14.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                vm.t("aboutThisTest"),
                fontSize = 15.sp,
                fontWeight = FontWeight.SemiBold,
            )
            Icon(
                Icons.Filled.KeyboardArrowDown,
                contentDescription = if (expanded) "Collapse" else "Expand",
                tint = c.gray,
                modifier = Modifier.size(20.dp).rotate(chevronRotation),
            )
        }

        AnimatedVisibility(
            visible = expanded,
            enter = expandVertically() + fadeIn(),
            exit = shrinkVertically() + fadeOut(),
        ) {
            Column(
                modifier = Modifier.padding(horizontal = 14.dp).padding(bottom = 14.dp),
            ) {
                state.source?.let { src ->
                    val manualUrl = state.verification?.manualUrl
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text(
                            "${vm.t("sourceLabel")}: ",
                            fontSize = 13.sp,
                            color = c.gray,
                        )
                        Text(
                            src,
                            fontSize = 13.sp,
                            color = if (manualUrl != null) c.blue else MaterialTheme.colorScheme.onSurface,
                            fontWeight = if (manualUrl != null) FontWeight.SemiBold else FontWeight.Normal,
                            modifier = if (manualUrl != null) {
                                Modifier.clickable {
                                    context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(manualUrl)))
                                }
                            } else {
                                Modifier
                            },
                        )
                    }
                }

                state.verification?.edition?.let { ed ->
                    Spacer(Modifier.height(6.dp))
                    Text(
                        "${vm.t("editionLabel")}: $ed",
                        fontSize = 13.sp,
                        color = c.gray,
                    )
                }

                Spacer(Modifier.height(6.dp))
                val officialLangs = state.officialTestLanguages
                val officialLangsText = if (officialLangs != null) {
                    formatOfficialLanguages(officialLangs, vm.t("officialTestLangsAndOthers"))
                } else {
                    vm.t("officialTestLangsUnknown", mapOf("agency" to state.agency))
                }
                Text(
                    "${vm.t("officialTestLangsLabel")}: $officialLangsText",
                    fontSize = 13.sp,
                    color = c.gray,
                )
                if (state.languages.size > 1) {
                    Spacer(Modifier.height(4.dp))
                    Text(
                        vm.t("officialTestPracticeNote"),
                        fontSize = 12.sp,
                        color = c.gray,
                    )
                }

                state.verification?.let { v ->
                    val hasBadges = v.precisionGrade != null ||
                        v.precisionAvgFidelity != null ||
                        v.recallCoveragePct != null
                    if (hasBadges) {
                        Spacer(Modifier.height(10.dp))
                        Row(
                            horizontalArrangement = Arrangement.spacedBy(6.dp),
                            modifier = Modifier.fillMaxWidth(),
                        ) {
                            v.precisionGrade?.let { grade ->
                                VerificationBadge(
                                    "${vm.t("gradeLabel")} $grade",
                                    c.green,
                                    c.greenLight,
                                )
                            }
                            v.precisionAvgFidelity?.let { fid ->
                                val fidStr = if (fid % 1.0 == 0.0) "${fid.toInt()}" else "%.1f".format(fid)
                                VerificationBadge(
                                    "$fidStr/10 ${vm.t("fidelityLabel")}",
                                    c.green,
                                    c.greenLight,
                                )
                            }
                            v.recallCoveragePct?.let { cov ->
                                val covStr = if (cov % 1.0 == 0.0) "${cov.toInt()}" else "%.1f".format(cov)
                                VerificationBadge(
                                    "$covStr% ${vm.t("coverageLabel")}",
                                    c.green,
                                    c.greenLight,
                                )
                            }
                        }
                    }

                    v.translations?.takeIf { it.isNotEmpty() }?.let { trans ->
                        Spacer(Modifier.height(8.dp))
                        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                            trans.forEach { (lang, verdict) ->
                                val isPassing = verdict == "PASS"
                                Row(
                                    verticalAlignment = Alignment.CenterVertically,
                                    horizontalArrangement = Arrangement.spacedBy(2.dp),
                                ) {
                                    Icon(
                                        if (isPassing) Icons.Filled.Check else Icons.Filled.Close,
                                        contentDescription = verdict,
                                        tint = if (isPassing) c.green else c.red,
                                        modifier = Modifier.size(14.dp),
                                    )
                                    Text(
                                        "${lang.uppercase()} ${verdict.lowercase()}",
                                        fontSize = 11.sp,
                                        fontWeight = FontWeight.SemiBold,
                                        color = if (isPassing) c.green else c.red,
                                    )
                                }
                            }
                        }
                    }
                }

                state.categories?.takeIf { it.isNotEmpty() }?.let { cats ->
                    Spacer(Modifier.height(12.dp))
                    val maxCount = cats.values.maxOrNull() ?: 1
                    cats.entries.sortedByDescending { it.value }.forEach { (cat, count) ->
                        CategoryRow(cat, count, maxCount, vm)
                        Spacer(Modifier.height(4.dp))
                    }
                }
            }
        }
    }
}

@Composable
private fun VerificationBadge(
    text: String,
    fg: Color,
    bg: Color,
) {
    Text(
        text,
        fontSize = 11.sp,
        fontWeight = FontWeight.SemiBold,
        color = fg,
        modifier = Modifier
            .clip(RoundedCornerShape(20.dp))
            .background(bg)
            .padding(horizontal = 8.dp, vertical = 3.dp),
    )
}

@Composable
private fun CategoryRow(
    cat: String,
    count: Int,
    maxCount: Int,
    vm: QuizViewModel,
) {
    val c = AppTheme.colors
    val label = cat.split("_").joinToString(" ") { word ->
        word.replaceFirstChar { it.uppercase() }
    }
    Row(
        modifier = Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(
            label,
            fontSize = 12.sp,
            color = c.gray,
            modifier = Modifier.width(130.dp),
        )
        Box(
            modifier = Modifier
                .weight(1f)
                .height(6.dp)
                .clip(RoundedCornerShape(3.dp))
                .background(c.grayLight),
        ) {
            Box(
                modifier = Modifier
                    .fillMaxHeight()
                    .fillMaxWidth(count.toFloat() / maxCount)
                    .clip(RoundedCornerShape(3.dp))
                    .background(c.blue),
            )
        }
        Spacer(Modifier.width(8.dp))
        Text(
            "$count",
            fontSize = 12.sp,
            fontWeight = FontWeight.SemiBold,
            color = c.blue,
            style = TextStyle(fontFeatureSettings = "tnum"),
        )
    }
}

@Composable
private fun ModeButton(
    title: String,
    desc: String,
    icon: ImageVector,
    isActive: Boolean,
    badgeCount: Int = 0,
    modifier: Modifier = Modifier,
    onClick: () -> Unit,
) {
    val c = AppTheme.colors
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(12.dp))
            .background(if (isActive) c.blueLight else c.card)
            .border(2.dp, if (isActive) c.blue else MaterialTheme.colorScheme.outline, RoundedCornerShape(12.dp))
            .clickable(onClick = onClick)
            .padding(12.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Icon(
            icon,
            contentDescription = null,
            tint = if (isActive) c.blue else c.gray,
            modifier = Modifier.size(24.dp),
        )
        Spacer(Modifier.height(4.dp))
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(
                title,
                fontSize = 14.sp,
                fontWeight = FontWeight.SemiBold,
                color = if (isActive) c.blue else MaterialTheme.colorScheme.onSurface,
            )
            if (badgeCount > 0) {
                Text(
                    "$badgeCount",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    modifier = Modifier
                        .clip(RoundedCornerShape(10.dp))
                        .background(c.red)
                        .padding(horizontal = 6.dp, vertical = 1.dp),
                )
            }
        }
        Text(desc, fontSize = 11.sp, color = if (isActive) c.blue else c.gray)
    }
}

private fun formatOfficialLanguages(
    languages: List<String>,
    andOthersText: String,
): String {
    val named = languages.filter { !it.equals("many", ignoreCase = true) }
    val hasMany = languages.any { it.equals("many", ignoreCase = true) }
    return when {
        named.isNotEmpty() && hasMany -> "${named.joinToString(", ")}, $andOthersText"
        named.isNotEmpty() -> named.joinToString(", ")
        hasMany -> andOthersText
        else -> ""
    }
}

File: ios/DriversTest/DriversTest/view/screen/HomeScreen.swift
import SwiftUI

struct HomeScreen: View {
    @ObservedObject var vm: QuizViewModel
    @ObservedObject var localizer: Localizer

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                LanguageBarView(localizer: localizer, availableLangs: vm.currentState?.languages)
                    .padding(.bottom, 4)

                Text(localizer.localized("title", substitutions: [
                    "state": vm.currentState?.code.uppercased() ?? "",
                    "state_name": vm.currentState?.name ?? "",
                ]))
                .font(.system(size: 28, weight: .bold))
                .foregroundColor(AppTheme.blue)

                Text(localizer.localized("subtitle", substitutions: [
                    "state_name": vm.currentState?.name ?? "",
                    "agency": vm.currentState?.agency ?? "",
                ]))
                .font(.system(size: 15))
                .foregroundColor(AppTheme.gray)

                if let state = vm.currentState {
                    Text(localizer.localized("passingScore", substitutions: [
                        "pass_pct": "\(state.passingScorePct)",
                        "pass_count": "\(state.passCount)",
                        "test_count": "\(state.testQuestionCount)",
                    ]))
                    .font(.system(size: 13))
                    .foregroundColor(AppTheme.gray)
                }

                // Stats banner
                if !vm.quizHistory.isEmpty {
                    StatsBannerView(vm: vm, localizer: localizer)
                }

                // Mode selector
                HStack(spacing: 8) {
                    ModeButton(
                        title: localizer.localized("modeRandom"),
                        desc: localizer.localized("modeRandomDesc"),
                        systemImage: "shuffle",
                        isActive: vm.quizMode == .random
                    ) { vm.quizMode = .random }

                    ModeButton(
                        title: localizer.localized("modeWeak"),
                        desc: localizer.localized("modeWeakDesc"),
                        systemImage: "target",
                        isActive: vm.quizMode == .weak,
                        badgeCount: vm.weakQuestions.count
                    ) { vm.quizMode = .weak }
                }

                // Count selector
                Text(localizer.localized("numQuestions"))
                    .font(.system(size: 15, weight: .semibold))

                HStack(spacing: 8) {
                    ForEach(vm.countOptions, id: \.self) { count in
                        let isAll = count == vm.currentState?.totalQuestions
                        Button {
                            vm.selectedCount = count
                        } label: {
                            Text(isAll ? localizer.localized("all") : "\(count)")
                                .font(.system(size: 16, weight: .semibold))
                                .padding(.horizontal, 20)
                                .padding(.vertical, 10)
                                .foregroundColor(vm.selectedCount == count ? AppTheme.onPrimary : AppTheme.blue)
                                .background(vm.selectedCount == count ? AppTheme.blue : AppTheme.card)
                                .cardStyle(borderColor: AppTheme.blue)
                                .frame(minHeight: 44)
                                .contentShape(Rectangle())
                        }
                    }
                }

                // Start button
                let weakEmpty = vm.quizMode == .weak && vm.weakQuestions.isEmpty
                Button {
                    vm.startQuiz()
                } label: {
                    Text(weakEmpty ? localizer.localized("noWeakSpots") : localizer.localized("startQuiz"))
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(AppTheme.onPrimary)
                        .frame(maxWidth: .infinity)
                        .padding(16)
                        .background(weakEmpty ? AppTheme.blue.opacity(0.5) : AppTheme.blue)
                        .clipShape(RoundedRectangle(cornerRadius: 12))
                }
                .disabled(weakEmpty)

                // About this test
                if let state = vm.currentState {
                    AboutTestSection(state: state, localizer: localizer)
                }

                // Change state
                Button {
                    vm.goStatePicker()
                } label: {
                    Text(localizer.localized("changeState"))
                        .font(.system(size: 14))
                        .foregroundColor(AppTheme.gray)
                }

            }
            .padding(.horizontal, 16)
            .padding(.top, 24)
        }
        .background(AppTheme.grayLight)
    }
}

// MARK: - About This Test

private struct AboutTestSection: View {
    let state: StateInfo
    @ObservedObject var localizer: Localizer
    @State private var isExpanded = false

    var body: some View {
        DisclosureGroup(isExpanded: $isExpanded) {
            VStack(alignment: .leading, spacing: 12) {
                if let verification = state.verification {
                    VerificationBadgesView(verification: verification, localizer: localizer)
                }

                if let source = state.source {
                    SourceRow(source: source, manualUrl: state.verification?.manualUrl, localizer: localizer)
                }

                if let categories = state.categories, !categories.isEmpty {
                    CategoryBreakdownView(categories: categories, localizer: localizer)
                }
            }
            .padding(.top, 8)
        } label: {
            HStack(spacing: 6) {
                Image(systemName: "info.circle")
                    .foregroundColor(AppTheme.blue)
                Text(localizer.localized("aboutThisTest"))
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundColor(.primary)
            }
        }
        .padding(14)
        .cardStyle(borderWidth: 1)
    }
}

private struct SourceRow: View {
    let source: String
    let manualUrl: String?
    @ObservedObject var localizer: Localizer

    var body: some View {
        HStack(alignment: .top, spacing: 6) {
            Text(localizer.localized("source") + ":")
                .font(.system(size: 13))
                .foregroundColor(AppTheme.gray)

            if let urlString = manualUrl, let url = URL(string: urlString) {
                Link(destination: url) {
                    Text(source)
                        .font(.system(size: 13))
                        .foregroundColor(AppTheme.blue)
                        .underline()
                        .multilineTextAlignment(.leading)
                }
            } else {
                Text(source)
                    .font(.system(size: 13))
                    .foregroundColor(.primary)
                    .multilineTextAlignment(.leading)
            }
        }
    }
}

private struct VerificationBadgesView: View {
    let verification: StateVerification
    @ObservedObject var localizer: Localizer

    private var isPassing: Bool {
        verification.overall == "PASS"
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            // Adaptive grid wraps badge pills; the iOS 16 Layout protocol is
            // unavailable on this app's iOS 15 target.
            LazyVGrid(
                columns: [GridItem(.adaptive(minimum: 96), spacing: 6, alignment: .leading)],
                alignment: .leading,
                spacing: 6
            ) {
                if let grade = verification.precisionGrade {
                    BadgePill(
                        label: localizer.localized("grade"),
                        value: grade,
                        systemImage: isPassing ? "checkmark.seal" : nil
                    )
                }

                if let fidelity = verification.precisionAvgFidelity {
                    BadgePill(
                        label: localizer.localized("fidelity"),
                        value: String(format: "%.1f/10", fidelity),
                        systemImage: fidelity >= 7.0 ? "checkmark.seal" : nil
                    )
                }

                if let coverage = verification.recallCoveragePct {
                    BadgePill(
                        label: localizer.localized("coverage"),
                        value: String(format: "%.0f%%", coverage),
                        systemImage: coverage >= 80 ? "checkmark.seal" : nil
                    )
                }

                if let translations = verification.translations, !translations.isEmpty {
                    let allPass = translations.values.allSatisfy { $0 == "PASS" }
                    BadgePill(
                        label: localizer.localized("translations"),
                        value: allPass ? "OK" : "partial",
                        systemImage: allPass ? "checkmark.seal" : nil
                    )
                }
            }

            HStack(spacing: 16) {
                if let edition = verification.edition {
                    HStack(spacing: 4) {
                        Text(localizer.localized("edition") + ":")
                            .foregroundColor(AppTheme.gray)
                        Text(edition)
                    }
                    .font(.system(size: 12))
                }

                if let verifiedAt = verification.verifiedAt {
                    HStack(spacing: 4) {
                        Text(localizer.localized("verified") + ":")
                            .foregroundColor(AppTheme.gray)
                        Text(Self.formatDate(verifiedAt))
                    }
                    .font(.system(size: 12))
                }
            }
        }
    }

    private static func formatDate(_ iso: String) -> String {
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        if let date = formatter.date(from: iso) {
            let display = DateFormatter()
            display.dateStyle = .medium
            display.timeStyle = .none
            return display.string(from: date)
        }
        formatter.formatOptions = [.withInternetDateTime]
        if let date = formatter.date(from: iso) {
            let display = DateFormatter()
            display.dateStyle = .medium
            display.timeStyle = .none
            return display.string(from: date)
        }
        return String(iso.prefix(10))
    }
}

private struct BadgePill: View {
    let label: String
    let value: String
    var systemImage: String?

    var body: some View {
        HStack(spacing: 4) {
            if let systemImage {
                Image(systemName: systemImage)
                    .font(.system(size: 11))
                    .foregroundColor(AppTheme.green)
            }
            Text(label)
                .foregroundColor(AppTheme.gray)
            Text(value)
                .fontWeight(.semibold)
        }
        .font(.system(size: 12))
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .background(AppTheme.blueLight)
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }
}

private struct CategoryBreakdownView: View {
    let categories: [String: Int]
    @ObservedObject var localizer: Localizer

    private var sortedCategories: [(name: String, count: Int)] {
        categories.map { (name: $0.key, count: $0.value) }
            .sorted { $0.count > $1.count }
    }

    private var maxCount: Int {
        categories.values.max() ?? 1
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            ForEach(sortedCategories, id: \.name) { cat in
                HStack(spacing: 8) {
                    Text(cat.name.replacingOccurrences(of: "_", with: " ").capitalized)
                        .font(.system(size: 12))
                        .frame(width: 120, alignment: .leading)
                        .lineLimit(1)

                    GeometryReader { geo in
                        RoundedRectangle(cornerRadius: 3)
                            .fill(AppTheme.blue)
                            .frame(width: geo.size.width * CGFloat(cat.count) / CGFloat(maxCount), height: 8)
                    }
                    .frame(height: 8)

                    Text("\(cat.count)")
                        .font(.system(size: 12, weight: .semibold))
                        .monospacedDigit()
                        .foregroundColor(AppTheme.gray)
                        .frame(width: 30, alignment: .trailing)
                }
            }
        }
    }
}
